tests/asap_testobject.py

```diff
@@ -0,0 +1,50 @@
+"""
+TestCase object to be used by the testing framework of the library
+
+"""
+
+import unittest
```

Comment:
Nice work with the unit tests! Would highly recommend switching to pytest (fast becoming the industry standard) for future unit testing:

https://docs.pytest.org/en/latest/

It makes tests a lot more compact because they can be written as Python functions as opposed to classes. I'll try to add some examples over the next couple of days!

Reply:
this has been done now, thanks for the recommendation